.github/workflows/deploy.yml

```diff
@@ -0,0 +1,75 @@
+name: Deploy
+
+on:
+  release:
+    types: [created]
+
+defaults:
+  run:
+    shell: pwsh
+
+env:
+  baseVersion: 1.0.0
+
+jobs:
+  build-and-test:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - name: Setup .NET
+      uses: actions/setup-dotnet@v2
+      with:
+        dotnet-version: | 
+            2.1.x
+            3.1.x
+            6.x
+    - name: Restore dependencies
+      run: dotnet restore
+    - name: Set Version
+      run: |
+        echo "${{ github.ref }}"
+        if ("${{ github.ref }}".startsWith("refs/tags/v")) {
+          $tagVersion = "${{ github.ref }}".substring(11)
+          echo "buildVersion=$tagVersion.${{ github.run_number }}" | Out-File -FilePath $env:GITHUB_ENV -Encoding utf8 -Append
+          echo "nugetVersion=$tagVersion" | Out-File -FilePath $env:GITHUB_ENV -Encoding utf8 -Append
+          echo "preRelease=false" | Out-File -FilePath $env:GITHUB_ENV -Encoding utf8 -Append
+        } else {
+          echo "buildVersion=${{ env.baseVersion }}.${{ github.run_number }}" | Out-File -FilePath $env:GITHUB_ENV -Encoding utf8 -Append
+          echo "nugetVersion=${{ env.baseVersion }}-ci${{ github.run_number }}" | Out-File -FilePath $env:GITHUB_ENV -Encoding utf8 -Append
+        }
+    - name: Build
+      run: dotnet build -p:Version=${{ env.buildVersion }} -p:ContinuousIntegrationBuild=True --no-restore --configuration Release
+    - name: Test
+      run: dotnet test --no-build --configuration Release --verbosity normal
+    - name: Pack
+      if: startsWith(github.ref, 'refs/tags/v')
+      run: dotnet pack -p:PackageVersion=${{ env.nugetVersion }} --configuration Release -o ${{env.DOTNET_ROOT}}/IntelliTect.UtilitiesPack --no-build 
+    - name: Upload Artifacts
+      if: startsWith(github.ref, 'refs/tags/v')
+      uses: actions/upload-artifact@v2
+      with:
+        name: NuGet
+        path: ${{env.DOTNET_ROOT}}/IntelliTect.UtilitiesPack
+      
+  deploy:
+    if: startsWith(github.ref, 'refs/tags/v')
+    runs-on: ubuntu-latest
+    needs: build-and-test
+    environment:
+      name: 'Production'
+    steps:
+      - name: Download artifact from build job
+        uses: actions/download-artifact@v2
+        with:
+          name: NuGet
+      - name: Push NuGet
+        run: |
+          $tagVersion = "${{ github.ref }}".substring(11)
+          echo "tagVersion=$tagVersion" | Out-File -FilePath $env:GITHUB_ENV -Encoding utf8 -Append
```

Comment:
Any reason this is doing environment variables rather than just a step output?

Reply:
Not sure I follow, sorry

Reply:
This appears to be setting an environment variable so that it can be consumed in the "Upload nupkg to Releases" step below. Typically passing variable between steps is done with [output parameters](https://docs.github.com/en/enterprise-server@3.4/actions/using-workflows/workflow-commands-for-github-actions#setting-an-output-parameter)

Reply:
Ahhh, How's this?
```yml
- name: Push NuGet
        run: |
          $tagVersion = "${{ github.ref }}".substring(11)
          echo '::set-output name=TAG_VERSION::$tagVersion'
          dotnet nuget push IntelliTect.Utilities.$tagVersion.nupkg --source https://api.nuget.org/v3/index.json -k ${{ secrets.NUGET_API_KEY }} --skip-duplicate
        id: tag-version
      - name: Upload nupkg to Releases
        uses: softprops/action-gh-release@v1
        with:
          fail_on_unmatched_files: true
          generate_release_notes: true
          files: IntelliTect.Utilities.${{ steps.tag-version.outputs.TAG_VERSION }}.nupkg
```

Reply:
lgtm.